### ISSUE TITLE
pyOpenMS get_massql_df() ValueError if a spectrum contains no peaks

### ISSUE THREAD

**Issue by precl** [OpenMS/OpenMS#5831]
The MSExperimentDF.get_massql_df() function raises a ValueError if a one of the MSSpektrum objects does not contain any peaks. It should instead just ignore these spectra. MS spectra without peaks can occur especially in MS 2 spectra if the fragmentation does not result fragments within the m/z range of the spectrometer.

Code to reproduce the bug:
https://gist.github.com/precl/d795885179403c1b4577e663ff660276

Traceback:
https://gist.github.com/precl/ab6662c03d80bab32d542718117c59e5

pyOpenMS 2.8.0dev Build: py39h9b97cb2_3  on Ubuntu 21.10
Installed from https://anaconda.org/openms/pyopenms

The offending function is np.amax() in line 370 of dataframes.py, np.amax raises a ValueError if the array is empty.
https://github.com/OpenMS/OpenMS/blob/develop/src/pyOpenMS/pyopenms/dataframes.py#L370
I was able to fix the bug with a small change.
`data = (inty, inty / np.amax(inty, initial=0), inty / np.sum(inty), mz, scan_num + 1, spec.getRT() / 60, _get_polarity(spec))`
`initial=0` adds 0 to the array, allowing np.amax to return 0 if it is empty.

If we cannot assume that intensities are always >= 0 if propose this fix:
```
try:
    data = (inty, inty / np.amax(inty), inty / np.sum(inty), mz, scan_num + 1, spec.getRT() / 60, _get_polarity(spec))
except ValueError:
    if not inty.shape == (0,):
        raise
    # if a spectrum does not contain any peaks
    # this sometimes happens with MS level 2 spectra
    data = (inty, inty / np.amax(inty, initial=0), inty / np.sum(inty), mz, scan_num + 1, spec.getRT() / 60, _get_polarity(spec))
```
Can we assume that intensities are always >= 0?

I am currently working on a pull request, but as this is a first for me it may take some time.

**Comment by axelwalter** [comment]
Thanks for pointing out that error and the elegant solution! We can assume the intensities to be >= 0 so setting the initial value in np.amax should work.

**Comment by axelwalter** [comment]
solved by #5957
